Bulk search & delete

andrew-nowak opened the pull request:
## What does this change?

- Improves image search command to include pagination to arbitrary depth, and outputting to file.
- Improves image delete command to allow hard-deletion
- Adds bulk image deletion command to delete all images according to a file containing a list of image ids.


make sure not to test against a profile pointing to prod!! the cli has very little protections from mistaken use

andrew-nowak committed:
paginated searching

andrew-nowak committed:
add option to output search results to file

andrew-nowak committed:
add bulk delete command

twrichards commented:
LOL at this feature (command line thumbnails)
![image](https://user-images.githubusercontent.com/19289579/189311111-b74ad801-6125-4e85-b3ab-3342def16c8c.png)

twrichards commented:
@andrew-nowak I think the README needs regenerating as it doesn't have the same stuff as the --help output, e.g. 
![image](https://user-images.githubusercontent.com/19289579/189313120-7a462bb2-27a4-4f2f-940d-bd41860b5807.png)
...where I wanted to know how to use the new paging functionality.

andrew-nowak committed:
update README.md

andrew-nowak commented:
Thanks! fix pushed

On Fri, 9 Sept 2022 at 10:00, Tom Richards ***@***.***> wrote:

> @andrew-nowak <https://github.com/andrew-nowak> I think the README needs
> regenerating as it doesn't have the same stuff as the --help output, e.g.
> [image: image]
> <https://user-images.githubusercontent.com/19289579/189313120-7a462bb2-27a4-4f2f-940d-bd41860b5807.png>
> ...where I wanted to know how to use the new paging functionality.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/guardian/grid-cli/pull/36#issuecomment-1241701560>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACTUQZULNLAI63T773OVKWLV5L4BBANCNFSM6AAAAAAQCJPSDI>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

-- 

This e-mail and all attachments are confidential and may also be 
privileged. If you are not the named recipient, please notify the sender 
and delete the e-mail and all attachments immediately. Do not disclose the 
contents to another person. You may not use the information for any 
purpose, or store, or copy, it in any way.  Guardian News & Media Limited 
is not liable for any computer viruses or other material transmitted with 
or as part of this e-mail. You should employ virus checking software.
 
Guardian News & Media Limited is a member of Guardian Media Group plc. 
Registered Office: PO Box 68164, Kings Place, 90 York Way, London, N1P 2AP. 
 Registered in England Number 908396

twrichards reviewed (approved):
Amazing work @andrew-nowak  👏

Locally, I've tested all three of the features in the description - worked perfectly. Logic looks 👌.

Thanks for fixing the readme 🎉

andrew-nowak committed:
ensure compilation before readme generation

If there's been updates to the typescript files without a following tsc, then the generated readme may be missing the new commands/flags
